splinepy/rational_bezier.py

```diff
@@ -0,0 +1,301 @@
+
+        Parameters
+        -----------
+        None
+
+        Returns
+        --------
+        None
+        """
+        if (
+            self.dim is None
+            or self.para_dim is None
+            or self.degrees is None
+            or self.control_points is None
+            or self.weights is None
+        ):
+            logging.debug(
+                "Spline - Not enough information to update cpp spline. "
+                "Skipping update or removing existing backend spline."
+            )
+            if hasattr(self, "_c_spline"):
+                delattr(self, "_c_spline")
+            return None
+
+        c_spline_class = f"RationalBezier{self.para_dim}P{self.dim}D"
+        self._c_spline = eval(c_spline_class)(
+            degrees=self.degrees,
+            control_points=self.control_points,
+            weights=self.weights
+        )
+
+        logging.debug(f"Spline - Your spline is {self.whatami}.")
+
+    def _update_p(self,):
+        """
+        Reads cpp spline and writes it here.
+        Probably get an error if cpp isn't ready for this.
+
+        Parameters
+        -----------
+        None
+
+        Returns
+        --------
+        None
+        """
+        # Don't use setters here, since it calls update_c each time
+        # and it causes us to lose array references.
+        self._degrees = self._c_spline.degrees
+        self._control_points = self._c_spline.control_points
+        self._weights = self._c_spline.weights
+        # silent setter
+        self._para_dim = self._c_spline.para_dim
+        self._dim = self._c_spline.dim
+        logging.debug(
+            "Spline - Updated python spline. CPP spline and python spline are "
+            "now identical."
+        )
+
+    def sample(self, resolution):
+        """
+        Overwrites basic sample routine by explicitly giving sampling location.
+
+        Parameters
+        -----------
+        resolution: int or list
+
+        Returns
+        --------
+        sampled: (prod(resolution), dim) np.ndarray
+        """
+        if isinstance(resolution, int):
+            resolution = [resolution for _ in range(self.para_dim)]
+
+        elif isinstance(resolution, (list or np.ndarray)):
+            if len(resolution) != self.para_dim:
+                raise ValueError(
+                    "Invalid resolution length. Should match para_dim"
+                )
+
+        else:
+            raise TypeError("resolution only accept list, np.ndarray or int.")
+
+        q = utils.raster_points(
+            [
+                [0. for _ in range(self.para_dim)],
+                [1. for _ in range(self.para_dim)],
+            ],
+            resolution
+        )
+        return self.evaluate(q)
+
+    def __mul__(self, factor):
+        """
+        Overloads multiplication between splines with different types of
+        degrees
+
+        The resulting spline fulfills the equation
+          factora(t) * factorb(t) = result(t)
```

Comment:
does this mean 
`factor_a.evaluate(t) * factor_b.evaluate(t) = result.evaluate(t)`?

Reply:
yes